outputs.tf

```diff
@@ -1,5 +1,5 @@
 output "public_ip" {
-  value = "${aws_eip.default.public_ip}"
+  value = "${coalesce(aws_eip.default.public_ip, aws_instance.default.public_ip)}"
```

Comment:
This is a great fix, but it does not fix `public_dns`. What should we do about that?

Reply:
If `aws_eip` resource is not being created, it may cause TF failure, because `aws_eip.default.public_ip` variable can't be resolved.
It should be tested.